runtime-modules/proposals/codex/src/lib.rs

```diff
@@ -750,18 +751,26 @@ impl<T: Config> Module<T> {
                         Error::<T>::InvalidFundingRequestProposalRepeatedAccount
                     );
 
-                    ensure!(
-                        funding_request.amount != Zero::zero(),
-                        Error::<T>::InvalidFundingRequestProposalBalance
-                    );
-
-                    ensure!(
-                        funding_request.amount <= T::FundingRequestProposalMaxAmount::get(),
-                        Error::<T>::InvalidFundingRequestProposalBalance
-                    );
-
                     visited_accounts.insert(account);
                 }
+
+                let total_funding_amount = funding_requests
+                    .iter()
+                    .try_fold(BalanceOf::<T>::zero(), |sum, el| {
+                        sum.checked_add(&el.amount)
+                    })
+                    .ok_or("Arithmetic Error")?;
+
+                // ensure total_funding_amount in (0, MAX]
+                ensure!(
+                    !total_funding_amount.is_zero(),
+                    Error::<T>::InvalidFundingRequestProposalBalance
+                );
```

Comment:
I think it's actually worth to check for non-zero amount on each account, not just `total_funding_amount.is_zero()`

Reply:
if $x_1, x_2, \dots, x_n$ are nonnegative integers and $x_1 + x_2 + \dots + x_n = 0$ then $x_1 = x_2 = \dots = x_n = 0$